integration_tests/test_standalone_grpc_server.py

```diff
@@ -0,0 +1,291 @@
+from rasa_sdk.grpc_errors import ResourceNotFound, ResourceNotFoundType
+from rasa_sdk.grpc_py import action_webhook_pb2_grpc, action_webhook_pb2
+from rasa_sdk.grpc_server import GRPC_ACTION_SERVER_NAME, _initialise_grpc_server
+from rasa_sdk.interfaces import ActionNotFoundException, ActionMissingDomainException
+from integration_tests.conftest import server_cert, server_cert_key, ca_cert, client_key, client_cert
+
+GRPC_PORT = 6000
+GRPC_TLS_PORT = 6001
+GRPC_HOST = 'localhost'
+
+
+@pytest.fixture(scope="module")
+def mock_executor() -> AsyncMock:
+    """Create a mock action executor."""
+    return AsyncMock(spec=ActionExecutor)
+
+
+@pytest.fixture(scope="module")
+def grpc_action_server(mock_executor: AsyncMock) -> None:
+    """Create a gRPC server for the action server."""
+
+    _event = asyncio.Event()
+
+    async def _run_grpc_server() -> None:
+        _grpc_action_server = _initialise_grpc_server(
+            action_executor=mock_executor,
+            port=GRPC_PORT,
+            max_number_of_workers=2,
+        )
+        await _grpc_action_server.start()
+        await _event.wait()
+        await _grpc_action_server.stop(None)
+
+    thread = threading.Thread(target=asyncio.run, args=(_run_grpc_server(),), daemon=True)
+    thread.start()
+    yield
+    _event.set()
+
+
+@pytest.fixture(scope="module")
+def grpc_tls_action_server(mock_executor: AsyncMock) -> None:
+    """Create a gRPC server for the action server."""
+
+    _event = asyncio.Event()
+
+    async def _run_grpc_server() -> None:
+        _grpc_action_server = _initialise_grpc_server(
+            action_executor=mock_executor,
+            port=GRPC_TLS_PORT,
+            max_number_of_workers=2,
+            ssl_server_cert=server_cert(),
+            ssl_server_cert_key=server_cert_key(),
+        )
+        await _grpc_action_server.start()
+        await _event.wait()
+        await _grpc_action_server.stop(None)
+
+    thread = threading.Thread(target=asyncio.run, args=(_run_grpc_server(),), daemon=True)
+    thread.start()
+    yield
+    _event.set()
+
+
+@pytest.fixture
+def grpc_channel() -> grpc.Channel:
+    """Create a gRPC channel for the action server."""
+    return grpc.insecure_channel(target=f"{GRPC_HOST}:{GRPC_PORT}", compression=grpc.Compression.Gzip)
+
+
+@pytest.fixture
+def grpc_action_client(grpc_channel: grpc.Channel) -> action_webhook_pb2_grpc.ActionServiceStub:
+    """Create a gRPC client for the action server."""
+    client = action_webhook_pb2_grpc.ActionServiceStub(grpc_channel)
+    return client
+
+
+@pytest.fixture
+def grpc_tls_channel() -> grpc.Channel:
+    """Create a gRPC channel for the action server."""
+    credentials = grpc.ssl_channel_credentials(
+        root_certificates=ca_cert(),
+        private_key=client_key(),
+        certificate_chain=client_cert(),
+    )
+    return grpc.secure_channel(
+        target=f"{GRPC_HOST}:{GRPC_TLS_PORT}",
```

Comment:
Should the host here be `GRPC_TLS_HOST`?

Reply:
We are running two servers on the localhost at different ports (6000 and 6001) so the host part is `localhost` for both. It's the port which is the differentiator.
Or is it harder to read the code? LMK

Reply:
Ah I see. Got confused by the fact that I thought `GRPC_TLS_HOST` should be paired with `GRPC_TLS_PORT`.